types/lodash/common/object.d.ts

```diff
@@ -1022,14 +1022,16 @@ declare module "../index" {
 
     type GetIndexedField<T, K> = K extends keyof T
         ? T[K]
-        : K extends `${number}`
-            ? 'length' extends keyof T
-                ? number extends T['length']
+        : K extends `${infer U extends number}`
```

Comment:
Does it matter that `1.23` and `.123` will work for this? It's fair if it doesn't matter since the previous code did not check against this, but it's possible to verify the result is an integer using this:
```suggestion
        : [K, K] extends [`${bigint}`, `${infer U extends number}`]
```